### ISSUE TITLE
ActiveRecord 6.1.7.7 test suite fails with `LoadError`

### ISSUE THREAD

**Issue by arielvalentin** [open-telemetry/opentelemetry-ruby-contrib#939]
Looks like there is a conflict with `sqlite3` in the test suite:

<https://github.com/open-telemetry/opentelemetry-ruby-contrib/actions/runs/8791124619/job/24124660982#step:3:172>

```console
  /opt/hostedtoolcache/Ruby/3.3.0/x64/lib/ruby/gems/3.3.0/gems/bundler-2.5.9/lib/bundler/rubygems_integration.rb:237:in `block (2 levels) in replace_gem': Error loading the 'sqlite3' Active Record adapter. Missing a gem it depends on? can't activate sqlite3 (~> 1.4), already activated sqlite3-2.0.1-x86_64-linux-gnu. Make sure all dependencies are added to Gemfile. (LoadError)
```

**Comment by xuan-cao-swi** [comment]
Activerecord [sqlite3_adapter](https://github.com/rails/rails/blob/v7.1.3.2/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb) is using `sqlite3  '~> 1.4'`, but sqlite3-ruby make installing the latest sqlite3 which cause conflict. 

sqlite3-ruby is out-dated anyway, and it changes (its name) to sqlite3; I guess it needs to change the [Gemfile](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/instrumentation/active_record/Gemfile#L15) to `sqlite3, '~> 1.4'`, and add `sqlite-dev \` in [Dockerfile](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/Dockerfile) to make sure sqlite3 gem can be installed.

**Comment by arielvalentin** [comment]
Fixed by [fix: resolve active_record testing issue #944](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/944)
